app/src/main/java/com/example/evently/ui/common/ConfirmFragmentNoInput.java

```diff
@@ -0,0 +1,112 @@
+package com.example.evently.ui.common;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.GradientDrawable;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.FrameLayout;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+
+import com.example.evently.R;
+
+public class ConfirmFragmentNoInput extends DialogFragment {
+
+    static final String headerKey = "header";
+    static final String messageKey = "message";
+    static final String cancelKey = "cancel";
+    static final String fragmentKey = "fragmentClass";
+
+    public ConfirmFragmentNoInput() {}
+
+    public static ConfirmFragmentNoInput newInstance(String headerText, String messageText, String cancelText, Class<? extends Fragment> fragmentClass) {
+        ConfirmFragmentNoInput confirmFragment = new ConfirmFragmentNoInput();
+        Bundle args = new Bundle();
+        args.putString(headerKey, headerText);
+        args.putString(messageKey, messageText);
+        args.putString(cancelKey, cancelText);
+        args.putString(fragmentKey, fragmentClass.getCanonicalName());
```

Comment:
I wouldn't recommend trying to put a class name in a bundle. Dynamic class loading is weird and error prone and this sort of reflection has been gradually getting removed.